api/v1alpha1/recipe_types.go

```diff
@@ -70,15 +75,34 @@ type Group struct {
 	// Whether to include any cluster-scoped resources. If nil or true, cluster-scoped resources are
 	// included if they are associated with the included namespace-scoped resources
 	IncludeClusterResources *bool `json:"includeClusterResources,omitempty"`
+	// Selects namespaces by label
+	IncludedNamespacesByLabel *metav1.LabelSelector `json:"includedNamespacesByLabel,omitempty"`
 	// List of namespaces to include.
 	//+optional
 	IncludedNamespaces []string `json:"includedNamespaces,omitempty"`
+	// List of namespace to exclude
+	ExcludedNamespaces []string `json:"excludedNamespaces,omitempty"`
+	// RestoreStatus restores status if set to all the includedResources specified. Specify '*' to restore all statuses for all the CRs
+	RestoreStatus *GroupRestoreStatus `json:"restoreStatus,omitempty"`
 	// Defaults to true, if set to false, a failure is not necessarily handled as fatal
 	Essential *bool `json:"essential,omitempty"`
+	// Whether to overwrite resources during restore. Default to false.
+	RestoreOverwriteResources *bool `json:"restoreOverwriteResources,omitempty"`
+}
+
+// GroupRestoreStatus is within resource groups which instructs velero to restore status for specified resources types, * would mean all
+type GroupRestoreStatus struct {
+	// List of resource types to include. If unspecified, all resource types are included.
+	IncludedResources []string `json:"includedResources,omitempty"`
+	// List of resource types to exclude.
+	ExcludedResources []string `json:"excludedResources,omitempty"`
 }
 
 // Workflow is the sequence of actions to take
 type Workflow struct {
+	// Name of recipe. Names "backup" and "restore" are reserved and implicitly used by default for
+	// backup or restore respectively
+	Name string `json:"name"`
```

Comment:
I'm going to assume an enum based validation was not helpful here?

Reply:
It is useful but limiting. Only those listed in the enum will be accepted thereby preventing users from creating other workflows.